clarity/src/vm/types/mod.rs

```diff
@@ -1287,6 +1289,16 @@ impl From<TupleData> for Value {
     }
 }
 
+impl From<ContractName> for Value {
+    fn from(name: ContractName) -> Self {
+        // ContractName is guaranteed to be between 5 and 40 bytes and contains only printable
+        // ASCII already, so this conversion should not fail.
+        Value::Sequence(SequenceData::String(CharType::ASCII(ASCIIData {
+            data: name.as_str().as_bytes().to_vec(),
+        })))
+    }
+}
```

Comment:
It may end up being more verbose, but I think it would add more clarity to the conversion to split this into two conversions:

```suggestion
impl From<ASCIIData> for Value {
    fn from(ascii: ASCIIData) -> Self {
        Value::Sequence(SequenceData::String(CharType::ASCII(ascii)))
    }
}
impl From<ContractName> for ASCIIData {
    fn from(name: ContractName) -> Self {
        // ContractName is guaranteed to be between 5 and 40 bytes and contains only printable
        // ASCII already, so this conversion should not fail.
        ASCIIData {
            data: name.as_str().as_bytes().to_vec(),
        }
    }
}
```

Then, instead of `Value::from(name)`, callers would need to perform `Value::from(ASCIIData::from(name))`. I know this seems like it is an unnecessary inflation of the code, but I don't think it's obvious that `Value::from(name)` should produce the ASCII variant.

Reply:
Fixed